src/store/reducer.js

```diff
@@ -0,0 +1,118 @@
+import {
+  SEARCH,
+  FILTER_MOVIE,
+  VISIBLE,
+  LOAD_MOVIES,
+  FILTER_ID,
+  REMOVE_MOVIE,
+  NEW_MOVIE,
+  ADD_MOVIE,
+  RESET,
+  LOAD_MOVIES_BY_TITLE,
+  LOAD_MOVIES_BY_DATE,
+  LOAD_MOVIES_BY_RATING,
+} from "./actionTypes";
+
+const initialState = {
+  movies: [],
+  searchFilter: "",
+  filterMovie: "",
+  movieArray: [],
+  visibility: true,
+  newMovie: {
+    title: "",
+    poster_path: "",
+    vote_average: "",
+    tagline: "",
+    genres: "",
+    overview: "",
+    release_date: "",
+    runtime: "",
+  },
+  id: {
+    title: "",
+    poster_path: "",
+    vote_average: "",
+    tagline: "",
+    genres: "",
+    overview: "",
+    release_date: "",
+    runtime: "",
+  },
+};
+
+const rootReducer = (state = initialState, action) => {
+  switch (action.type) {
+    case LOAD_MOVIES:
+      return {
+        ...state,
+        movies: [action.payload],
+      };
+    case LOAD_MOVIES_BY_TITLE:
+      return {
+        ...state,
+        movies: [action.payload],
+      };
+
+    case LOAD_MOVIES_BY_DATE:
+      return {
+        ...state,
+        movies: [action.payload],
+      };
+    case LOAD_MOVIES_BY_RATING:
+      return {
+        ...state,
+        movies: [action.payload],
```

Comment:
A lot of actions that do the same thing with state, no need to duplicate them, just do the following. Or you can just use LOAD_MOVIES in all this cases.
```
switch (action.type) {
     case LOAD_MOVIES:
     case LOAD_MOVIES_BY_TITLE:
     case LOAD_MOVIES_BY_DATE:
     case LOAD_MOVIES_BY_RATING:
       return {
         ...state,
         movies: [action.payload],
      };
```

Reply:
Fixed